client/packages/programs/src/JsonForms/components/PeriodSearch.tsx

```diff
@@ -43,6 +44,21 @@ const UIComponent = (props: ControlProps) => {
     }
   );
 
+  const periodId = extractProperty(core?.data, 'periodId');
+  useMemo(() => {
+    if (periodId && !period) {
+      const period = data?.pages
+        ?.find(page => page.data.nodes.some(period => period.id === periodId))
+        ?.data.nodes.find(period => period.id === periodId);
+      if (period) {
+        setPeriod(period);
+      }
+    }
```

Comment:
Having to use nested searching like this almost makes me want to have separate usePeriodList hooks (one for infinateQuery, one just returning nodes).

But I imagine this remains the most streamlined way to manage this.